MediaBrowser.Controller/Entities/BaseItem.cs

```diff
@@ -1481,7 +1499,10 @@ private async Task<bool> RefreshLocalTrailers(IHasTrailers item, MetadataRefresh
 
         private async Task<bool> RefreshExtras(BaseItem item, MetadataRefreshOptions options, List<FileSystemMetadata> fileSystemChildren, CancellationToken cancellationToken)
         {
-            var newExtras = LoadExtras(fileSystemChildren, options.DirectoryService).Concat(LoadThemeVideos(fileSystemChildren, options.DirectoryService)).Concat(LoadThemeSongs(fileSystemChildren, options.DirectoryService));
+            var newExtras = LoadExtras(fileSystemChildren, options.DirectoryService)
```

Comment:
The LINQ `Concat` method is terribly slow. Since all the functions return an array the total length is know and you could create a new array and use `.CopyTo`.